addon/-private/array.ts

```diff
@@ -147,16 +73,94 @@ class TrackedArray<T = unknown> {
     thisArg?: unknown
   ): TrackedArray<T> | TrackedArray<U> {
     return mapfn
-      ? createArrayProxy(Array.from(iterable, mapfn, thisArg))
-      : createArrayProxy(Array.from(iterable));
+      ? new TrackedArray(Array.from(iterable, mapfn, thisArg))
+      : new TrackedArray(Array.from(iterable));
   }
 
   static of<T>(...arr: T[]): TrackedArray<T> {
-    return createArrayProxy(arr);
+    return new TrackedArray(arr);
   }
 
   constructor(arr: T[] = []) {
-    return createArrayProxy(arr.slice());
+    let clone = arr.slice();
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    let self = this;
```

Comment:
I'm using this blast from the past because not all of the proxy hooks have a receiver? or at least my editor didn't say that they did.

If I'm wrong, I'll gladly swap out all the self references to receiver.

Reply:
Well, for the array case they all do - because the only hooks needed are `get` and `set`. So there you can use `receiver`. But for the object case you're right.

Reply:
Actually no, you can't use `receiver` as it's the Proxy object, not the instance of the class that has the needed methods.

Reply:
that's why `let self = this` 🙃